cmd/zc_traverser_local.go

```diff
@@ -122,33 +165,69 @@ func WalkWithSymlinks(fullPath string, walkFunc filepath.WalkFunc, followSymlink
-					walkQueue = append(walkQueue, walkItem{
-						fullPath:     result,
-						relativeBase: computedRelativePath,
-					})
+				rStat, err := os.Stat(result)
+				if err != nil {
+					glcm.Info(fmt.Sprintf("Failed to get properties of symlink target at %s: %s", result, err))
+					return nil
+				}
+
+				if rStat.IsDir() {
+					if !seenPaths.HasSeen(result) {
+						seenPaths.Record(result)
+						seenPaths.Record(slPath) // Note we've seen the symlink as well. We shouldn't ever have issues if we _don't_ do this because we'll just catch it by symlink result
+						walkQueue = append(walkQueue, walkItem{
+							fullPath:     result,
+							relativeBase: computedRelativePath,
+						})
+						// enumerate the FOLDER now (since its presence in seenDirs will prevent its properties getting enumerated later)
+						return walkFunc(common.GenerateFullPath(fullPath, computedRelativePath), symlinkTargetFileInfo{rStat, fileInfo.Name()}, fileError)
+					} else {
+						glcm.Info(fmt.Sprintf("Ignored already linked directory pointed at %s (link at %s)", result, common.GenerateFullPath(fullPath, computedRelativePath)))
+					}
 				} else {
-					glcm.Info(fmt.Sprintf("Ignored already linked directory pointed at %s (link at %s)", result, common.GenerateFullPath(fullPath, computedRelativePath)))
+					glcm.Info(fmt.Sprintf("Symlinks to individual files are not currently supported, so will ignore file at %s (link at %s)", result, common.GenerateFullPath(fullPath, computedRelativePath)))
+					// TODO: remove the above info call and enable the below, with suitable multi-OS testing
+					//    including enable the test: TestWalkWithSymlinks_ToFile
+					/*
+							// It's a symlink to a file. Just process the file because there's no danger of cycles with links to individual files.
+							// (this does create the inconsistency that if there are two symlinks to the same file we will process it twice,
+							// but if there are two symlinks to the same directory we will process it only once. Because only directories are
+							// deduped to break cycles.  For now, we are living with the inconsistency. The alternative would be to "burn" more
+							// RAM by putting filepaths into seenDirs too, but that could be a non-trivial amount of RAM in big directories trees).
+
+							// TODO: this code here won't handle the case of (file-type symlink) -> (another file-type symlink) -> file
+						    //    But do we WANT to handle that?  (since it opens us to risk of file->file cycles, and we are deliberately NOT
+						    //    putting files in our map, to reduce RAM usage).  Maybe just detect if the target of a file symlink its itself a symlink
+						    //    and skip those cases with an error message?
+							// Make file info that has name of source, and stats of dest (to mirror what os.Stat calls on source will give us later)
```

Comment:
Actually, maybe instead of putting files in the map, we could just temporarily hold a map for this particular operation and then discard it.

Reply:
Let's review for Files in 10.4.1, as discussed.